public/app/features/query/state/DashboardQueryRunner/UnifiedAlertStatesWorker.ts

```diff
@@ -54,53 +59,53 @@ export class UnifiedAlertStatesWorker implements DashboardQueryRunnerWorker {
     }
 
     const { dashboard } = options;
-    return from(
-      getBackendSrv().get(
-        '/api/prometheus/grafana/api/v1/rules',
-        {
-          dashboard_uid: dashboard.uid,
-        },
-        `dashboard-query-runner-unified-alert-states-${dashboard.id}`
-      )
-    ).pipe(
-      map((result: PromRulesResponse) => {
-        if (result.status === 'success') {
-          this.hasAlertRules[dashboard.uid] = false;
-          const panelIdToAlertState: Record<number, AlertStateInfo> = {};
-          result.data.groups.forEach((group) =>
-            group.rules.forEach((rule) => {
-              if (isAlertingRule(rule) && rule.annotations && rule.annotations[Annotation.panelID]) {
-                this.hasAlertRules[dashboard.uid] = true;
-                const panelId = Number(rule.annotations[Annotation.panelID]);
-                const state = promAlertStateToAlertState(rule.state);
+    const fetchData = async () => {
+      const promRules = dispatch(
+        alertRuleApi.endpoints.prometheusRuleNamespaces.initiate({
+          ruleSourceName: GRAFANA_RULES_SOURCE_NAME,
+          dashboardUid: dashboard.uid,
+        })
+      );
+      return (await promRules).data;
+    };
 
-                // there can be multiple alerts per panel, so we make sure we get the most severe state:
-                // alerting > pending > ok
-                if (!panelIdToAlertState[panelId]) {
-                  panelIdToAlertState[panelId] = {
-                    state,
-                    id: Object.keys(panelIdToAlertState).length,
-                    panelId,
-                    dashboardId: dashboard.id,
-                  };
-                } else if (
-                  state === AlertState.Alerting &&
-                  panelIdToAlertState[panelId].state !== AlertState.Alerting
-                ) {
-                  panelIdToAlertState[panelId].state = AlertState.Alerting;
-                } else if (
-                  state === AlertState.Pending &&
-                  panelIdToAlertState[panelId].state !== AlertState.Alerting &&
-                  panelIdToAlertState[panelId].state !== AlertState.Pending
-                ) {
-                  panelIdToAlertState[panelId].state = AlertState.Pending;
-                }
+    const res: Observable<PromRuleGroupDTO[]> = from<Promise<RuleNamespace[]>>(fetchData()).pipe(
```

Comment:
nit: instead of typing the variable here and the `from<>` I would add a return type to `async fetchData(): Promise<RuleNamespace[]>` and let the type inference handle this bit.